### PR TITLE
405 migrate to materialui 5

### LINKED ISSUE #405: Migrate to Material UI version 5
https://mui.com/material-ui/migration/migration-v4/

### LINKED ISSUE #482: Refactor text styles
We should refactor all text styles on our wallet. 

**Design Material** https://www.figma.com/file/ZTqF54s6CSHTwAxp0U0JjQ/Circles-myxogastria?node-id=4673%3A9745

### LINKED ISSUE #626: Hyperlinks in popup and left side menu should have more prominent color (probably white)
Screenshot from 2023-03-08 13-06-19

### PR DESCRIPTION
**Pull request by mikozet**
part which I didn't figure out yet is why font-weight changes are not being visible - it's probably due to wrongly uploading fonts but I do not see mistake inc code currently

### CODE REVIEW

**Comment by mikozet** [comment]
There have been lot of little changes and that task requires full check of the app. Mostly in terms of how app is looking as logical part hasn't been touched

File: src/index.html [hunk 1]
```diff
@@ -6,9 +6,28 @@
...
+      href="https://joincircles.net/wp-content/uploads/2021/10/favicon.png"
+    />
+    <meta
+      name="msapplication-TileImage"
+      content="https://joincircles.net/wp-content/uploads/2021/10/favicon.png"
+    />
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <style>
+      body {
```
Comment (mikozet) [inline src/index.html:23]: temporary `body` here, will be moved once we gather styles for specific text types

**Comment by louilinn** [comment]
in dev-preview

File: src/index.html [hunk 2]
```diff
@@ -6,9 +6,28 @@
...
+        font-size: 0.875rem;
+        font-weight: 400;
+        line-height: 1.43;
```
Comment (louilinn) [inline src/index.html:26]: Try to have it in css file

**Comment by louilinn** [comment]
I have now added all the issues that this PR resolves. Please always make sure this is done. 

I also discovered that the behavior of pink shadow has changed / improved in that is disappears when you reach the bottom of the page.

These are the types of things you need to include in a checklist.

**Review by louilinn — COMMENTED** [review]
Problems I found so far (negative changes requires fixing):
- floating menu ends up in front of "My Wallets" when clicking this option
- pink shadow is not visable in scrollable views until scrolling has started
- loading indicator wheel for maximum amount in send view has changed color from pink to grey

Other changes that I noticed that are good / unproblematic (no change required)
- pink shadow stops showing when bottom of scroll is reached (better behaviour than before)
- badge in tabs is no longer grayer in unselected tabs (this was unintended before)

**Comment by mikozet** [comment]
I didn't notice difference with pink shadow thus that there has been no checklist..
Current updates:
- [x] Fix pink shadow - could you write device details while testing? I didn't register pink shadow not showing during page load when there is scroll. I adapt it so it behaves better when in reaches end of scroll to dissappear (which didn't happend always on my device (desktop, linux, chrome).
- [x] Fix loading spinner color on transfer view
- [x] Fix Floating menu ends up in front of "My Wallets" when clicking this option
- [x] Fix button for closing floating menu does not appear
- [x] I noticed an issue with icons not being visible on toolbar during scroll - currently on production as well
- [x] Change color links in popups (Learn more) have wrong color (currently bug in production)
- [x] Change color menu links in foter from left side menu (currently bug in production)
- [x] Fix no internet notification without icon

Screenshot from 2023-03-29 13-29-36
Screenshot from 2023-03-29 15-27-23
Screenshot from 2023-03-29 15-33-17

File: src/index.html (hunk 2 above)
Comment (mikozet) [reply src/index.html:26]: I would leave it as it is now - I didn't manage to properly link css file so it takes effect

File: src/components/TransferInfoCard.js [hunk 3]
```diff
@@ -68,7 +71,12 @@ const TransferInfoCard = ({
               <Typography className={classes.cardHeaderContent} component="div">
                 <CirclesLogoSVG height={fontSize} width={fontSize} />
                 <span>{text}</span>
-                {isLoading && <CircularProgress size={fontSize} />}
+                {isLoading && (
+                  <CircularProgress
+                    classes={{ svg: classes.svg }}
```
Comment (louilinn) [inline src/components/TransferInfoCard.js]: Have you checked if other Circular Progress are purple or grey? 
This feels like an adhoc way to fix the problem. I think all Circular progress should be purple not just this one. Do we know why the color has changed in this upgrade?

File: src/index.js [hunk 4]
```diff
@@ -22,10 +22,12 @@ if (window.ethereum) {
 const Root = (props) => (
   <Provider store={props.store}>
     <NoSsr>
-      <ThemeProvider theme={theme}>
-        <CssBaseline />
-        <App />
-      </ThemeProvider>
+      <StyledEngineProvider injectFirst>
```
Comment (louilinn) [inline src/index.js:25]: removes the need for "important!"
good!

File: src/store/onboarding/actions.js [hunk 5]
```diff
@@ -197,6 +197,7 @@ export function finalizeNewAccount() {
       notify({
         text: <WelcomeMessage />,
         type: NotificationsTypes.SPECIAL,
+        lifetime: 259200, // 3 days
```
Comment (louilinn) [inline src/store/onboarding/actions.js]: mistake for testing

File: src/styles/theme.js [hunk 6]
```diff
@@ -125,16 +126,7 @@ export default createTheme({
...
     },
   },
-  props: {
-    MuiDialog: {
-      maxWidth: 'lg',
-    },
-  },
   palette: {
-    background: {
-      default: colors.whiteAlmost,
```
Comment (louilinn) [inline src/styles/theme.js:135]: where is background color set now?

File: src/components/Navigation.js [hunk 7]
```diff
@@ -76,10 +80,7 @@ const useStyles = makeStyles((theme) => ({
     display: 'block',
     color: theme.palette.primary.contrastText,
   },
-  navigationLinkLabel: {
-    textTransform: 'initial',
-    fontWeight: theme.typography.fontWeightLight,
-  },
+  navigationLinkLabel: {},
```
Comment (louilinn) [inline src/components/Navigation.js]: why keep empty class?

File: src/components/ActivityStream.js [hunk 8]
```diff
@@ -80,7 +80,7 @@ const useStyles = makeStyles((theme) => ({
   },
   cardContent: {
     paddingTop: 0,
-    paddingBottom: `${theme.spacing(1.5)}px !important`,
+    paddingBottom: `${theme.spacing(1.5)} !important`,
```
Comment (louilinn) [inline src/components/ActivityStream.js]: do we need important?

File: src/components/Button.js [hunk 9]
```diff
@@ -50,13 +51,15 @@ const useStyles = makeStyles((theme) => ({
...
-      '& .MuiButton-label': {
+      // '& .MuiButton-label': {
+      '& .MuiTouchRipple-root': {
         background: 'none',
         color: theme.custom.colors.lola,
         '-webkit-text-fill-color': theme.custom.colors.lola,
       },
     },
-    '& .MuiButton-label': {
+    // '& .MuiButton-label': {
```
Comment (louilinn) [inline src/components/Button.js]: remove all these commented lines

File: src/components/TransferInfoCard.js (hunk 3 above)
Comment (mikozet) [reply src/components/TransferInfoCard.js]: correct

File: src/components/TabNavigationAction.js [hunk 10]
```diff
@@ -12,17 +12,22 @@ const useStyles = makeStyles((theme) => ({
...
+    '& g, & path': {
+      fill: theme.custom.colors.lily,
+    },
     '&:hover': {
       borderBottom: `1px solid ${theme.custom.colors.oldLavender}`,
-      '& .MuiTab-wrapper': {
-        color: theme.custom.colors.oldLavender,
-      },
+      color: theme.custom.colors.oldLavender,
+      // },
```
Comment (louilinn) [inline src/components/TabNavigationAction.js]: remove

**Review by louilinn — COMMENTED** [review]
Some more comments :)

File: src/components/ActivityStream.js (hunk 8 above)
Comment (louilinn) [reply src/components/ActivityStream.js]: check if it is also needed still in any places at all in the codebase

Comment (mikozet) [reply src/components/ActivityStream.js]: I did it another way :)

File: src/components/Button.js (hunk 9 above)
Comment (mikozet) [reply src/components/Button.js]: correct

File: src/styles/theme.js (hunk 6 above)
Comment (mikozet) [reply src/styles/theme.js:135]: I added it, as it looks like it was missed:
from old docs: https://v4.mui.com/customization/default-theme/#default-theme
new docs: https://mui.com/material-ui/customization/default-theme/#default-theme
by default it is color white and we had our color `whiteAlmost`

I removed `pallete.background.paper` as it was not used

File: src/components/TransferInfoCard.js [hunk 11]
```diff
@@ -67,8 +67,13 @@ const TransferInfoCard = ({
             <Tooltip arrow title={tooltip}>
               <Typography className={classes.cardHeaderContent} component="div">
                 <CirclesLogoSVG height={fontSize} width={fontSize} />
-                <span>{text}</span>
-                {isLoading && <CircularProgress size={fontSize} />}
+                <span className={classes.text}>{text}</span>
+                {isLoading && (
+                  <CircularProgress
+                    classes={{ svg: classes.svg }}
```
Comment (louilinn) [inline src/components/TransferInfoCard.js]: now this can be romoved? as styling of span was moved to text class?

**Comment by louilinn** [comment]
Regarding pink shadow I'm using Chromium 111.0.5563.110-1~deb11u1
I see a behavioral difference between circles.garden and the preview of this. It maybe isn't a big deal.
In firefox it is the exact opposite behavior for the two versions :sweat_smile: . So I think you can ignore this for now..

File: src/store/onboarding/actions.js (hunk 5 above)
Comment (mikozet) [reply src/store/onboarding/actions.js]: corrected

File: src/components/TabNavigationAction.js (hunk 10 above)
Comment (mikozet) [reply src/components/TabNavigationAction.js]: corrected

File: src/components/Navigation.js (hunk 7 above)
Comment (mikozet) [reply src/components/Navigation.js]: corrected

File: src/components/TransferInfoCard.js (hunk 11 above)
Comment (louilinn) [reply src/components/TransferInfoCard.js]: @mikozet ?

**Review by louilinn — APPROVED** [review]

**Pull request merged**